src/tribler/core/components/torrent_checker/torrent_checker/torrent_checker.py

```diff
@@ -208,6 +212,16 @@ def load_torrents_checked_from_db(self) -> Dict[bytes, HealthInfo]:
                                                   last_check=torrent.last_check, self_checked=True)
         return result
 
+    def schedule_check(self, torrents: list, loop: AbstractEventLoop) -> None:
+        """
+        Schedule a health check of the given torrents.
+
+        :type torrents: list[TorrentState]
+        """
+        for t in torrents:
+            loop.call_soon_threadsafe(self.register_anonymous_task, f"Check health {t.infohash}",
```

Comment:
This seems like it would necessitate a major refactoring of the torrent checker to solve sanely then. Secondly, I think that this refactoring would overlap too much with #7328.

Taking that into account, I'll close this PR in favor of another (future) more thorough refactoring PR.